### PR TITLE
GEODE-7584: refactor pdx handling

### PR DESCRIPTION
**Pull request by pdxcodemonkey**

### CODE REVIEW

**Comment by pdxcodemonkey** [comment]
@vfordpivotal @mreddington @davebarnes97 @dihardman @karensmolermiller

File: cppcache/include/geode/PdxWriter.hpp [hunk 1]
```diff
@@ -400,6 +400,11 @@ class APACHE_GEODE_EXPORT PdxWriter {
       std::shared_ptr<PdxUnreadFields> unread) = 0;
 
   virtual std::shared_ptr<PdxSerializer> getPdxSerializer() const = 0;
+
+  virtual int32_t getBytesWritten() { return m_bytesWritten; }
```
Comment (pivotal-jbarrett) [inline cppcache/include/geode/PdxWriter.hpp]: Why are we introducing this new method into the public API? If it is a size value it should be of `size_t`.

File: cppcache/integration/framework/Cluster.cpp [hunk 2]
```diff
@@ -317,8 +317,9 @@ apache::geode::client::Cache Cluster::createCache(
     cacheFactory.set(property.first, property.second);
   }
 
-  auto cache = cacheFactory.set("log-level", "none")
+  auto cache = cacheFactory.set("log-level", "debug")
```
Comment (pivotal-jbarrett) [inline cppcache/integration/framework/Cluster.cpp]: This should not be part of your PR. Test should not log. Tests should assert behavior. Logging is a last resort for diagnosing uncertain failures.

File: cppcache/src/PdxHelper.cpp [hunk 3]
```diff
@@ -47,92 +47,105 @@ PdxHelper::PdxHelper() {}
 
 PdxHelper::~PdxHelper() {}
 
-void PdxHelper::serializePdx(
-    DataOutput& output, const std::shared_ptr<PdxSerializable>& pdxObject) {
-  auto pdxII = std::dynamic_pointer_cast<PdxInstanceImpl>(pdxObject);
-  auto cacheImpl = CacheRegionHelper::getCacheImpl(output.getCache());
-  auto pdxTypeRegistry = cacheImpl->getPdxTypeRegistry();
-  auto& cachePerfStats = cacheImpl->getCachePerfStats();
+std::shared_ptr<PdxInstanceImpl> PdxHelper::getPdxInstance(
```
Comment (pivotal-jbarrett) [inline cppcache/src/PdxHelper.cpp]: I fear that hiding the `std::dynamic_pointer_cast` and the potential for it go return `nullptr` is going to make for issues later.

File: cppcache/src/PdxHelper.cpp [hunk 4]
```diff
@@ -172,8 +185,8 @@ std::shared_ptr<PdxSerializable> PdxHelper::deserializePdx(DataInput& dataInput,
       if (preserveData != nullptr) {
         pdxTypeRegistry->setPreserveData(
             pdxObjectptr, preserveData,
-            cacheImpl
-                ->getExpiryTaskManager());  // it will set data in weakhashmap
+            cacheImpl->getExpiryTaskManager());  // it will set data in
```
Comment (pivotal-jbarrett) [inline cppcache/src/PdxHelper.cpp]: Cleanup this comment.

File: cppcache/src/PdxLocalWriter.cpp [hunk 5]
```diff
@@ -330,11 +331,13 @@ PdxWriter& PdxLocalWriter::markIdentityField(const std::string&) {
   return *this;
 }
 
-uint8_t* PdxLocalWriter::getPdxStream(int& pdxLen) {
+uint8_t* PdxLocalWriter::getPdxStream() {
```
Comment (pivotal-jbarrett) [inline cppcache/src/PdxLocalWriter.cpp]: How is the caller supposed to know how big this buffer is?

File: cppcache/integration/framework/Cluster.cpp (hunk 2 above)
Comment (pdxcodemonkey) [reply cppcache/integration/framework/Cluster.cpp]: Meh - leftover from debugging test.  Will remove.

File: cppcache/include/geode/PdxWriter.hpp (hunk 1 above)
Comment (pdxcodemonkey) [reply cppcache/include/geode/PdxWriter.hpp]: Hmm, this was supposed to have been removed from PdxWriter, let me have a look and see what happened.  I think that "removes" it from the public API, but maybe only sort of....

As for why it's there: you and I had a conversation about this one when I first did this work, although that was admittedly a good long while ago.  The new method cleans up this particular bit of mental gymnastics:

```
    if (cacheImpl != nullptr) {	
      uint8_t* stPos = const_cast<uint8_t*>(output.getBuffer()) +	
                       ptc.getStartPositionOffset();	
      int pdxLen = PdxHelper::readInt32(stPos);	
      cachePerfStats.incPdxSerialization(	
          pdxLen + 1 + 2 * 4);  // pdxLen + 93 DSID + len + typeID	
    }
```

Comment (pivotal-jbarrett) [reply cppcache/include/geode/PdxWriter.hpp]: Hmm... yeah I don't recall. It should probably have some documentation associated with it so we know what the purpose is.

**Comment by pdxcodemonkey** [comment]
Closing this PR, since it's clear I'm not going to get around to addressing the changes requested any time soon, if at all.  GEODE-7584 in Jira can remain active as a low-priority issue indefinitely, I suppose.

**Pull request closed without merging**